// asposepdfcloud/src/models/annotationType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var AnnotationType;
(function (AnnotationType) {
    AnnotationType["Text"] = "Text";
    AnnotationType["Circle"] = "Circle";
    AnnotationType["Polygon"] = "Polygon";
    AnnotationType["PolyLine"] = "PolyLine";
    AnnotationType["Line"] = "Line";
    AnnotationType["Square"] = "Square";
    AnnotationType["FreeText"] = "FreeText";
    AnnotationType["Highlight"] = "Highlight";
    AnnotationType["Underline"] = "Underline";
    AnnotationType["Squiggly"] = "Squiggly";
    AnnotationType["StrikeOut"] = "StrikeOut";
    AnnotationType["Caret"] = "Caret";
    AnnotationType["Ink"] = "Ink";
    AnnotationType["Link"] = "Link";
    AnnotationType["Popup"] = "Popup";
    AnnotationType["FileAttachment"] = "FileAttachment";
    AnnotationType["Sound"] = "Sound";
    AnnotationType["Movie"] = "Movie";
    AnnotationType["Screen"] = "Screen";
    AnnotationType["Widget"] = "Widget";
    AnnotationType["Watermark"] = "Watermark";
    AnnotationType["TrapNet"] = "TrapNet";
    AnnotationType["PrinterMark"] = "PrinterMark";
    AnnotationType["Redaction"] = "Redaction";
    AnnotationType["Stamp"] = "Stamp";
    AnnotationType["RichMedia"] = "RichMedia";
    AnnotationType["Unknown"] = "Unknown";
    AnnotationType["PDF3D"] = "PDF3D";
})(AnnotationType = exports.AnnotationType || (exports.AnnotationType = {}));
